### PR TITLE
fix: remove all GSQL headers

### PR DESCRIPTION
**Pull request by olavloite**
The `/*GSQL*/` header will soon be unsupported and should be removed from
all clients.

### CODE REVIEW

File: .ci/e2e-expected/ddl-dml-batching.txt [hunk 1]
```diff
@@ -1 +1 @@
-ERROR: FAILED_PRECONDITION: Executing updates is not allowed for DDL batches. "CREATE TABLE testtable (id integer PRIMARY KEY, data integer); INSERT INTO testtable (id, data) VALUES (1, 2); INSERT INTO testtable (id, data) VALUES (3, 4);"
+ERROR: FAILED_PRECONDITION: Executing updates is not allowed for DDL batches. "CREATE TABLE testtable (id integer PRIMARY KEY, data integer); INSERT INTO testtable (id, data) VALUES (1, 2); INSERT INTO testtable (id, data) VALUES (3, 4)"
```
Comment (olavloite) [inline .ci/e2e-expected/ddl-dml-batching.txt:1]: This PR removes the PGAdapter specific parser for removing comments and uses the one in the Java client library as well. There is one small difference between the two; the one in the Java client library will remove any trailing semicolon from the SQL string. This does not semantically change the commands, as a trailing semicolon in a SQL statement is ignored.

File: src/main/resources/metadata/default_command_metadata.json [hunk 2]
```diff
@@ -1,81 +1,81 @@
...
+      "input_pattern": "SELECT n\\.nspname as \"[^\"]*\",[ \n]+c\\.relname as \"[^\"]*\",[ \n]+CASE c\\.relkind (WHEN '.' THEN '[^']*' )+END as \"[^\"]*\",[ \n]+pg_catalog\\.pg_get_userbyid\\(c\\.relowner\\) as \"[^\"]*\"\nFROM pg_catalog\\.pg_class c[ \n]+LEFT JOIN pg_catalog\\.pg_namespace n ON n\\.oid = c\\.relnamespace\nWHERE c\\.relkind IN \\('r'(?:,'p')?,'v','m','S','f',''\\)[ \n]+AND n\\.nspname <> 'pg_catalog'[ \n]+AND (?:n\\.nspname <> 'information_schema'[ \n]+AND )?n\\.nspname !~ '\\^pg_toast'[ \n]+AND (?:n\\.nspname <> 'information_schema'[ \n]+AND )?pg_catalog\\.pg_table_is_visible\\(c\\.oid\\)\nORDER BY 1,2;?$",
```
Comment (olavloite) [inline src/main/resources/metadata/default_command_metadata.json:4]: Here also: Any trailing semicolon in the input pattern is ignored.

File: src/main/resources/metadata/strict_command_metadata.json [hunk 3]
```diff
@@ -1,119 +0,0 @@
-{
```
Comment (olavloite) [inline src/main/resources/metadata/strict_command_metadata.json:1]: This entire file was unused.

File: src/test/java/com/google/cloud/spanner/pgadapter/JdbcSimpleModeMockServerTest.java [hunk 4]
```diff
@@ -157,7 +157,7 @@ public void testMixedBatch() throws SQLException {
         assertEquals(
             String.format(
                 "ERROR: FAILED_PRECONDITION: Executing updates is not allowed for DDL batches. \"%s\"",
-                sql),
+                sql.substring(0, sql.length() - 1)),
```
Comment (olavloite) [inline src/test/java/com/google/cloud/spanner/pgadapter/JdbcSimpleModeMockServerTest.java:163]: Ignore the trailing semicolon.

File: src/test/java/com/google/cloud/spanner/pgadapter/StatementParserTest.java [hunk 5]
```diff
@@ -68,67 +77,72 @@ public void testRemoveCommentsAndTrim() {
...
+    result = parser.removeCommentsAndTrim(sqlStatement);
     Assert.assertEquals(result, sqlStatement);
 
     sqlStatement = "select *\nfrom foo # this is a comment\nwhere bar=1";
-    result = StatementParser.removeCommentsAndTrim(sqlStatement);
+    result = parser.removeCommentsAndTrim(sqlStatement);
     Assert.assertEquals(result, sqlStatement);
 
     // When parameters are mixed with dollar-quoted string
-    sqlStatement = "$1$$?it$?s$$$2";
```
Comment (olavloite) [inline src/test/java/com/google/cloud/spanner/pgadapter/StatementParserTest.java:114]: This was not a valid PostgreSQL SQL fragment, as there is no space between the query parameter and the start/end of the dollar-quoted string. The PGAdapter parser accepted it, but it is rejected by PostgreSQL and by the parser in the Java client library.

This also applies to the SQL fragments below.

**Review by Vizerai — APPROVED** [review]
LGTM

**Review by snehashah16 — APPROVED** [review]

**Pull request merged**